src/Service.Tests/SqlTests/GraphQLMutationTests/GraphQLMutationTestBase.cs

```diff
@@ -677,6 +677,185 @@ public async Task DeleteMutationWithVariablesAndMappings(string dbQuery, string
             Assert.AreEqual(result.RootElement.GetProperty("count").GetInt64(), 0);
         }
 
+        /// <summary>
+        /// Performs concurrent update mutations on the same item and validates that the responses
+        /// returned are consistent
+        /// gQLMutation1 : Updates the title column of Book table to New Title
+        /// gQLMutation2 : Updates the title column of Book table to Updated Title
+        /// The title field in the responses returned for each of the mutations should be
+        /// the same value it had written to the table.
+        /// </summary>
+        [TestMethod]
+        public async Task TestParallelUpdateMutations()
+        {
+            string graphQLMutationName = "updatebook";
+            string gQLMutation1 = @"
+                mutation {
+                    updatebook(id : 1, item: { title: ""New Title"" })
+                    {
+                        title
+                    }
+                }";
+
+            string gQLMutation2 = @"
+                mutation {
+                    updatebook(id : 1, item: { title: ""Updated Title"" })
+                    {
+                        title
+                    }
+                }";
+
+            Task<JsonElement> responeTask1 = ExecuteGraphQLRequestAsync(gQLMutation1, graphQLMutationName, isAuthenticated: true);
```

Comment:
How can we guarantee these are triggered in parallel? Shouldn't we use 2 threads to trigger those?

Reply:
Running them concurrently using the async operator is sufficient to recreate this bug.

This is a snip by running this integration test without the changes needed for creating a transaction 
![Test Repro Screenshot](https://user-images.githubusercontent.com/11196553/235924811-ba9c25b3-ac7f-4b91-b7a6-f22f28604d05.png)

It could be possible that the two requests might not be landing at the same exact point in time but they seem to be landing with a negligible delta that still causes this inconsistent behavior for graphQL mutations.